fix: toggle of page header on rows

preda-bogdan opened the pull request:
### Summary
Improved the toggle for the page header.
Previously it would only mount on Global Page Header Settings.

### Will affect the visual aspect of the product
<!-- It includes visual changes? -->
NO

### Screenshots <!-- if applicable -->
![image](https://user-images.githubusercontent.com/23024731/192231032-b60ed4ff-3a74-4786-a311-c67918906fb6.png)


### Test instructions
<!-- Describe how this pull request can be tested. -->
1. Go to Customizer > Page Header > Page Header Top or Page Header Bottom
2. Check that the builder opens
3. Check the builder opens as before for the Header section

<!-- Issues that this pull request closes. -->
Closes Codeinwp/neve-pro-addon#2173.
<!-- Should look like this: `Closes #1, #2, #3.` . -->

abaicus commented on assets/apps/customizer-controls/src/builder/HFGBuilderComponent.tsx:
```diff
@@ -151,10 +155,14 @@ const HFGBuilderComponent: React.FC<Props> = ({ control, portalMount }) => {
 					if (
-						activeSection.id ===
-						'neve_pro_global_page_header_settings'
+						[
+							'neve_pro_global_page_header_settings',
+							'hfg_page_header_layout_top',
+							'hfg_page_header_layout_bottom',
+						].includes(activeSection.id)
 					) {
 						setMounted(true);
+						setHidden(false);
```
I don't think listening to sections and unhiding the page header is necessary. I'm not sure why this binding to the expanded section existed in the first place. There can be only one active panel, regardless of what section inside it is active.

abaicus commented on assets/apps/customizer-controls/src/builder/HFGBuilderComponent.tsx:
```diff
@@ -78,6 +78,10 @@ const HFGBuilderComponent: React.FC<Props> = ({ control, portalMount }) => {
 			.state('expandedPanel')
 			.bind((panel: Record<string, unknown>) => {
 				if (panel.id && panel.id === `hfg_${builder}` && isHidden) {
+					if (panel.id === `hfg_page_header`) {
+						setHidden(true);
+						return true;
+					}
```
I don't think this is necessary, as mentioned in the main review comment.

abaicus reviewed (commented):
@preda-bogdan 

Maybe there's something I am missing, but is there any reason for the default behavior change in [this condition](https://github.com/Codeinwp/neve/blob/152fea45aacfd9167b5a0ae068fcf12f57b92cd2/assets/apps/customizer-controls/src/builder/HFGBuilderComponent.tsx#L140), especially for the page header?

Removing that condition should make things work fine. Again, I'm not aware of the reason it was circumvented in [this commit](https://github.com/Codeinwp/neve/commit/e77285bc6b13e6226a4462d6f3b5b3620661ae27). Do you remember?

preda-bogdan commented:
@abaicus the page header should open on section enter, this was decided when we changed the global settings for the page header. I've fixed the issues you mentioned. The behavior is that the builder is opened after entering the sections instead of being opened from the panel that is the reasoning for that condition.

irinelenache commented:
@preda-bogdan Tested and the issue is fixed now, thank you 🚀

cristian-ungureanu commented:
@preda-bogdan if that's true, we should also close the builder after coming back from the section. Otherwise, I think we can just remove the condition that Andrei mentioned and be consistent. It's weird having the header / footer working in one way and the page header in another.
Do you remember the reasoning?

preda-bogdan commented:
@cristian-ungureanu I've now changed it so that it unmounts and hides the panel when exiting to the page header panel from the child sections.

irinelenache commented:
@preda-bogdan Now the header builder is opening only one time if i switch between `Global page header settings, Page header top and Page header bottom`. If i want to open it again, i have to go back to main board of customizer settings and access Page header again. Here is a video https://vertis.d.pr/I6lsfR

preda-bogdan commented:
@irinelenache Good catch, there was another issue introduced by this change, the panel would hide on components settings for the page header. I've reworked the logic, please check again when you have time and let me know if everything works as expected now. Thank you!

irinelenache commented:
@preda-bogdan Now everything works fine, thank you 🚀
